### PR TITLE
UCT/CM: If the random service_id is busy, generate another one.

### LINKED ISSUE #1259: ib_cm_listen() failed
http://e2e-gw.mellanox.com:4080/hpc/scrap/users/mtt/scratch/ucx_ompi/20170123_172127_11251_76700_clx-hercules-001/html/test_stdout_FuxW_V.txt
```
/hpc/local/benchmarks/hpcx_install_Sunday/hpcx-gcc-redhat7.2/ompi-v2.x/bin/mpirun \
-np 192 -mca btl_openib_warn_default_gid_prefix 0 --bind-to core --tag-output --timestamp-output \
--display-map -mca pml ucx -x UCX_SHM_DEVICES=all -x UCX_NET_DEVICES=mlx5_0:1 -x UCX_ACC_DEVICES=all \
-mca coll_hcoll_enable 0 -x UCX_IB_ADDR_TYPE=ib_local -x UCX_TLS=all \
-mca opal_pmix_base_async_modex 1 -mca mpi_add_procs_cutoff 0 -mca pmix_base_collect_data 0 \
--map-by node \
/hpc/scrap/users/mtt/scratch/ucx_ompi/20170123_172127_11251_76700_clx-hercules-001/installs/a5wh/tests/mpich_tests/mpich-mellanox.git/test/mpi/coll/coll13
...
Mon Jan 23 18:23:43 2017[1,103]<stdout>:[1485188623.398447] [clx-hercules-003:30937:0]       cm_iface.c:308  UCX  ERROR ib_cm_listen() failed: Device or resource busy
Mon Jan 23 18:23:43 2017[1,103]<stddiag>:[clx-hercules-003:30937] PML ucx cannot be selected
...
```

### PR DESCRIPTION
**Pull request by yosefe**
fixes #1259

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1359/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3302/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1361/ for details.

File: src/uct/ib/cm/cm_iface.c [hunk 1]
```diff
@@ -295,12 +293,25 @@ static UCS_CLASS_INIT_FUNC(uct_cm_iface_t, uct_md_h md, uct_worker_h worker,
...
+    do {
+        self->service_id = (uint32_t)(ucs_generate_uuid((uintptr_t)self) &
+                                      (~IB_CM_ASSIGN_SERVICE_ID_MASK));
+        ret = ib_cm_listen(self->listen_id, self->service_id, 0);
+        if (ret) {
+            if (errno == EBUSY) {
+                /* The generated service id is already in use - try to
+                 * generate another one.
+                 */
+                ucs_debug("ib_cm_listen(service_id=0x%x) failed: %m",
```
Comment (miked-mellanox) [inline src/uct/ib/cm/cm_iface.c]: s/failed/busy

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1364/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3305/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1369/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1371/ for details.

**Pull request merged**